From 40c6775ee39d1b95632e2894e90d3c1b978e5a36 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 6 Sep 2022 07:43:26 +0800
Subject: [PATCH 34/60] Revert "drm: bridge: adv7511: Add check for
 mipi_dsi_driver_register"

This reverts commit 6a43236ebcfba2fb2840e640b844b7fe211727d2.
---
 drivers/gpu/drm/bridge/adv7511/adv7511_drv.c | 17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
index 430c5e8f0388e..8bac392cab794 100644
--- a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
+++ b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
@@ -1381,21 +1381,10 @@ static struct i2c_driver adv7511_driver = {
 
 static int __init adv7511_init(void)
 {
-	int ret;
-
-	if (IS_ENABLED(CONFIG_DRM_MIPI_DSI)) {
-		ret = mipi_dsi_driver_register(&adv7533_dsi_driver);
-		if (ret)
-			return ret;
-	}
-
-	ret = i2c_add_driver(&adv7511_driver);
-	if (ret) {
-		if (IS_ENABLED(CONFIG_DRM_MIPI_DSI))
-			mipi_dsi_driver_unregister(&adv7533_dsi_driver);
-	}
+	if (IS_ENABLED(CONFIG_DRM_MIPI_DSI))
+		mipi_dsi_driver_register(&adv7533_dsi_driver);
 
-	return ret;
+	return i2c_add_driver(&adv7511_driver);
 }
 module_init(adv7511_init);
 
-- 
2.25.1

